src/zos-io.cc

```diff
@@ -758,6 +759,32 @@ int __open_orig(const char *filename, int opts, ...) asm("@@A00144");
 int __mkstemp_orig(char *) asm("@@A00184");
 FILE *__fopen_orig(const char *filename, const char *mode) asm("@@A00246");
 int __mkfifo_orig(const char *pathname, mode_t mode) asm("@@A00133");
+struct utmpx *__getutxent_orig(void) asm("getutxent");
+
+int utmpxname(char * file) {
+  char buf[PATH_MAX];
+  size_t file_len = strnlen(file, PATH_MAX);
```

Comment:
Off by one error. Either increase the buffer by one or change strnlen to return one less than you are

Reply:
Ah right, I changed it to strnlen(file, PATH_MAX - 1);

Reply:
Lgtm